client/src/NetworkClient.ts

```diff
@@ -68,16 +92,87 @@ export class NetworkClient {
         this.$iframe = await NetworkClient._createIframe(this._endpoint) as HTMLIFrameElement;
         const targetWindow = this.$iframe.contentWindow as Window;
         this._eventClient = await EventClient.create(targetWindow, NetworkClient.getAllowedOrigin(this._endpoint));
+
+        this.on(NetworkClient.Events.API_READY, () => this._apiLoadingState = 'ready');
+        this.on(NetworkClient.Events.API_FAIL, () => this._apiLoadingState = 'failed');
+        this.on(NetworkClient.Events.CONSENSUS_SYNCING, () => this._consensus = 'syncing');
+        this.on(NetworkClient.Events.CONSENSUS_ESTABLISHED, () => this._consensus = 'established');
+        this.on(NetworkClient.Events.CONSENSUS_LOST, () => this._consensus = 'lost');
+        this.on(NetworkClient.Events.PEERS_CHANGED, (peerCount: number) => this._peerCount = peerCount);
+        this.on(NetworkClient.Events.BALANCES_CHANGED,
+            (balances: Map<string, number>) => this._balances = balances);
+        this.on(NetworkClient.Events.TRANSACTION_PENDING,
+            (tx: Partial<DetailedPlainTransaction>) => this._pendingTransactions.set(tx.hash!, tx));
+        this.on(NetworkClient.Events.TRANSACTION_EXPIRED, (txHash: string) => {
+            this._expiredTransactions.push([this.headInfo.height, txHash]);
+            this._pendingTransactions.delete(txHash);
+        });
+        this.on(NetworkClient.Events.TRANSACTION_MINED, (tx: DetailedPlainTransaction) => {
+            this._minedTransactions.set(tx.hash, tx);
+            this._pendingTransactions.delete(tx.hash);
+        });
+        this.on(NetworkClient.Events.TRANSACTION_RELAYED, (tx: Partial<DetailedPlainTransaction>) => {
+            tx.blockHeight = this.headInfo.height;
+            this._relayedTransactions.set(tx.hash!, tx);
+        });
+        this.on(NetworkClient.Events.HEAD_CHANGE, (headInfo: { height: number, globalHashrate: number}) => {
+            this._headInfo = headInfo;
+            this._evictCachedTransactions();
+        });
+    }
+
+    public setAsDefaultInstance() {
+        if (NetworkClient._instance) throw new Error('Default Instance already exists!');
+        NetworkClient._instance = this;
     }
 
-    public async on(event: string, callback: EventCallback) {
+    public async on(event: NetworkClient.Events, callback: EventCallback) {
```

Comment:
Since the NetworkClient namespace is not used for anything else, the Events could go directly in the global namespace as `NetworkEvent` (note as well the singular here).

Reply:
I like the namespacing approach and it is also closer to what we use in our JS code.

Reply:
I think it's unnecessary code to write and to store and the `NetworkClient` namespace is not used for anything else.

Reply:
Resolved by https://github.com/nimiq/network/pull/3#discussion_r239180556.